psydac/api/fem.py

```diff
@@ -464,7 +464,10 @@ def assemble(self, *, reset=True, **kwargs):
             self._matrix.exchange_assembly_data()
 
         if self._matrix: self._matrix.ghost_regions_in_sync = False
-        return self._matrix
+        if self._matrix == None:
+            return None
+        else : 
+            return self._matrix.copy()
```

Comment:
I understand your reasoning here, but if we assemble the bilinear form only once (which is what we do all the time in our unit tests), we end up creating two matrices when we only need one. This means that the memory usage is twice what it should be.

CC: @saidctb